public/app/features/plugins/utils.ts

```diff
@@ -72,7 +71,10 @@ export function buildPluginSectionNav(
     }
 
     return setPageToActive(child, currentUrl);
-  });
+  }
+
+  // Find and set active page
+  copiedPluginNavSection.children = (copiedPluginNavSection?.children ?? []).map(findAndSetActivePage);
```

Comment:
one thing that came up a lot during development of nested folders and all of it's recursive functions was infinite loops from (my) dodgy code, which was really annoying to debug.

Given this is kind of dealing with third party data (nav models supplied by plugins?) do you think it might be worth guarding against infinite recursion here?

Reply:
maybe. 😅  how would you want to guard against that, hardcode a limit? 🤔 

i don't think it's possible for an _external_ plugin to create these deeply nested structures, pretty sure they're limited to just a flat list of pages. but it is possible for us internally to create structures with arbitrary depth.

Reply:
Yeah - it's pretty rudimentary. You maintain a counter that you pass through each call, and then limit it to like only 10 or 100 or 1000 iterations or whatever you think is appropriate.

Because this is using `.map` at the moment, it would require a bit of restructuring the code. 

Note that i ended up _not_ keeping this in nested folders, mainly because it is a bit of faff.

Up to you :)

Reply:
added it, don't _think_ the code needs restructuring that much with `.map`, but might have missed something 😅